src/main/resources/static/vulnerableApp.js

```diff
@@ -160,14 +169,15 @@ function handleElementAutoSelection(vulnerableAppEndPointData, id = 0) {
   }
 
   if (id === 0) {
-    detailTitle.innerHTML = vulnerableAppEndPointData[id]["Description"];
-  } else {
+// Assuming detailTitle is a DOM element and vulnerableAppEndPointData is an object with user-controlled data.
+// Replace innerHTML with textContent to prevent the browser from parsing the input as HTML.
+detailTitle.textContent = vulnerableAppEndPointData[id]["Description"];
     innerMaster.innerHTML = "";
   }
 
   vulnerabilitySelected = vulnerableAppEndPointData[id]["Name"];
-  detailTitle.innerHTML = vulnerableAppEndPointData[id]["Description"];
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities